src/craftdef.cpp

```diff
@@ -1020,11 +962,17 @@ class CCraftDefManager: public IWritableCraftDefManager
 		}
 		return os.str();
 	}
-	virtual void registerCraft(CraftDefinition *def)
+	virtual void registerCraft(CraftDefinition *def, IGameDef *gamedef)
 	{
 		verbosestream<<"registerCraft: registering craft definition: "
 				<<def->dump()<<std::endl;
 		m_craft_definitions.push_back(def);
+
+		CraftInput input;
+		ItemStack output;
+		output.deSerialize(def->getOutput(input, gamedef).item);
+
+		m_output_to_craft_definition_map[output.name].push_back(def);
```

Comment:
It would seem that these should be cleaned up in ::clear() in the same way m_craft_definitions is.